app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -196,13 +212,47 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote): Musi
 		}
 	}
 
+	/**
+	 * Retrieves the cover art Bitmap for the provided ImageUri. If it is present in the coverArtCache
+	 * then the Bitmap is returned otherwise the imagesApi is called to asynchronously add the cover
+	 * art Bitmap to the cache and returns null in the meantime.
+	 */
+	fun getCoverArt(imageUri: ImageUri): Bitmap? {
+		val coverArt = coverArtCache.get(imageUri)
+		if (coverArt != null) {
+			return coverArt
+		} else {
+			remote.imagesApi.getImage(imageUri, Image.Dimension.THUMBNAIL).setResultCallback { coverArtResponse ->
+				coverArtCache.put(imageUri, coverArtResponse)
+			}
+		}
+		return null
+	}
+
+	/**
+	 * Creates the QueueMetadata for the current queue containing the title, subtitle, queue songs,
+	 * and queue cover art.
+	 */
+	private fun buildQueueMetadata() {
+		val recentlyPlayedUri = "com.spotify.recently-played"
+		val li = ListItem(recentlyPlayedUri, recentlyPlayedUri, null, null, null, false, true)
+		remote.contentApi.getChildrenOfItem(li, 1, 0).setResultCallback { recentlyPlayed ->
+			val item = recentlyPlayed?.items?.get(0)
+			if (item != null) {
+				remote.imagesApi.getImage(item.imageUri, Image.Dimension.THUMBNAIL).setResultCallback { coverArt ->
+					queueMetadata = QueueMetadata(item.title,item.subtitle,SpotifyMusicMetadata.createSpotifyMusicMetadataList(this, queueItems),coverArt)
+				}
+			}
+		}
+	}
+
 	/**
 	 * Loads the complete children contents of the given parentItem
 	 * checking if stillValid at each step
 	 * and then passing it to onComplete at the end
 	 */
 	private fun loadPaginatedItems(parentItem: ListItem,
-	                               stillValid: () -> Boolean, onComplete: (List<MusicMetadata>) -> Unit) {
+								   stillValid: () -> Boolean, onComplete: (List<MusicMetadata>) -> Unit) {
```

Comment:
Keep these spaces in place to make sure it lines up with the parameters from the previous line

Reply:
Fixed this